src/lib.rs

```diff
@@ -626,6 +627,36 @@ impl MergePreference {
     is_bit_set!(is_fastforward_only, MergePreference::FASTFORWARD_ONLY);
 }
 
+
+bitflags! {
+    /// Check attribute flags: Reading values from index and working directory.
+    ///
+    /// When checking attributes, it is possible to check attribute files
+    /// in both the working directory (if there is one) and the index (if
+    /// there is one).  You can explicitly choose where to check and in
+    /// which order using the following flags
+    ///
+    /// Core git usually checks the working directory then the index,
+    /// except during a checkout when it checks the index first.  It will
+    /// use index only for creating archives or for a bare repo (if an
+    /// index has been specified for the bare repo).
+    pub struct CheckAttributeFlags: u32 {
```

Comment:
I'm not honestly sure exactly how these flags behave for libgit2, I think it's for how it decides which .gitattributes files to use, but it's a bit unclear, so I haven't added unit tests for them.